rln-wasm/src/lib.rs

```diff
@@ -20,6 +20,64 @@ pub struct RLNWrapper {
     instance: RLN<'static>,
 }
 
+// Macro to call_with_error_msg methods with arbitrary amount of arguments,
+// First argument to the macro is context,
+// second is the actual method on `RLNWrapper`
+// rest are all other arguments to the method
+macro_rules! call_with_error_msg {
+    ($instance:expr, $method:ident, $error_msg:expr $(, $arg:expr)*) => {
+        {
+            let new_instance: &mut RLNWrapper = $instance.process();
+            if new_instance.instance.$method($($arg.process()),*).is_ok() {
+                Ok(())
+            } else {
+                Err($error_msg)
+            }
+
+        }
+    }
+}
```

Comment:
if the method returns a result, we can bubble the error up. please combine the errors so that it is more meaningful

Reply:
Methods that can be used here return `bool` only, no `Result`s

Reply:
we use .is_ok() though, am i missing something?

Reply:
I was wrong, pointing to a function with the same name but different signature.

Fixed as requested

Reply:
Awesome, thanks!